File: models/Coding.js
const utils = require('../utils/utils');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}studyKey`,
                label: `Unique study key - [${labelPrefix}studyKey]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}siteName`,
                label: `Name of the site associated with the coded data - [${labelPrefix}siteName]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}siteId`,
                label: `Site ID associated with the coded data - [${labelPrefix}siteId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}subjectId`,
                label: `Subject ID associated with the coded data - [${labelPrefix}subjectId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}subjectKey`,
                label: `Subject key (display ID) associated with the coded data - [${labelPrefix}subjectKey]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}formId`,
                label: `Form ID where the coded data originates - [${labelPrefix}formId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}formName`,
                label: `Name of the form where the coded data originates - [${labelPrefix}formName]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}formKey`,
                label: `Form key where the coded data originates - [${labelPrefix}formKey]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}revision`,
                label: `Revision number of the coding entry - [${labelPrefix}revision]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}recordId`,
                label: `Record ID associated with the coded data - [${labelPrefix}recordId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}variable`,
                label: `Variable name (field) that was coded - [${labelPrefix}variable]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}value`,
                label: `Original value entered that required coding - [${labelPrefix}value]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}codingId`,
                label: `Mednet coding ID - [${labelPrefix}codingId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}code`,
                label: `Standardized code assigned (e.g., dictionary term) - [${labelPrefix}code]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}codedBy`,
                label: `Name of the user who performed the coding - [${labelPrefix}codedBy]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}reason`,
                label: `Reason for coding or any notes on changes - [${labelPrefix}reason]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}dictionaryName`,
                label: `Name of the dictionary used (e.g., MedDRA) - [${labelPrefix}dictionaryName]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}dictionaryVersion`,
                label: `Version of the dictionary used - [${labelPrefix}dictionaryVersion]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}dateCoded`,
                label: `Date when the coding was performed - [${labelPrefix}dateCoded]`,
                type: 'string',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'studyKey': bundle.inputData?.[`${keyPrefix}studyKey`],
            'siteName': bundle.inputData?.[`${keyPrefix}siteName`],
            'siteId': bundle.inputData?.[`${keyPrefix}siteId`],
            'subjectId': bundle.inputData?.[`${keyPrefix}subjectId`],
            'subjectKey': bundle.inputData?.[`${keyPrefix}subjectKey`],
            'formId': bundle.inputData?.[`${keyPrefix}formId`],
            'formName': bundle.inputData?.[`${keyPrefix}formName`],
            'formKey': bundle.inputData?.[`${keyPrefix}formKey`],
            'revision': bundle.inputData?.[`${keyPrefix}revision`],
            'recordId': bundle.inputData?.[`${keyPrefix}recordId`],
            'variable': bundle.inputData?.[`${keyPrefix}variable`],
            'value': bundle.inputData?.[`${keyPrefix}value`],
            'codingId': bundle.inputData?.[`${keyPrefix}codingId`],
            'code': bundle.inputData?.[`${keyPrefix}code`],
            'codedBy': bundle.inputData?.[`${keyPrefix}codedBy`],
            'reason': bundle.inputData?.[`${keyPrefix}reason`],
            'dictionaryName': bundle.inputData?.[`${keyPrefix}dictionaryName`],
            'dictionaryVersion': bundle.inputData?.[`${keyPrefix}dictionaryVersion`],
            'dateCoded': bundle.inputData?.[`${keyPrefix}dateCoded`],
        }
    },
}

File: client/scala-gatling/src/gatling/scala/org/openapitools/client/model/Coding.scala

package org.openapitools.client.model


case class Coding (
    /* Unique study key */
    _studyKey: Option[String],
    /* Name of the site associated with the coded data */
    _siteName: Option[String],
    /* Site ID associated with the coded data */
    _siteId: Option[Integer],
    /* Subject ID associated with the coded data */
    _subjectId: Option[Integer],
    /* Subject key (display ID) associated with the coded data */
    _subjectKey: Option[String],
    /* Form ID where the coded data originates */
    _formId: Option[Integer],
    /* Name of the form where the coded data originates */
    _formName: Option[String],
    /* Form key where the coded data originates */
    _formKey: Option[String],
    /* Revision number of the coding entry */
    _revision: Option[Integer],
    /* Record ID associated with the coded data */
    _recordId: Option[Integer],
    /* Variable name (field) that was coded */
    _variable: Option[String],
    /* Original value entered that required coding */
    _value: Option[String],
    /* Mednet coding ID */
    _codingId: Option[Integer],
    /* Standardized code assigned (e.g., dictionary term) */
    _code: Option[String],
    /* Name of the user who performed the coding */
    _codedBy: Option[String],
    /* Reason for coding or any notes on changes */
    _reason: Option[String],
    /* Name of the dictionary used (e.g., MedDRA) */
    _dictionaryName: Option[String],
    /* Version of the dictionary used */
    _dictionaryVersion: Option[String],
    /* Date when the coding was performed */
    _dateCoded: Option[String]
)
object Coding {
    def toStringBody(var_studyKey: Object, var_siteName: Object, var_siteId: Object, var_subjectId: Object, var_subjectKey: Object, var_formId: Object, var_formName: Object, var_formKey: Object, var_revision: Object, var_recordId: Object, var_variable: Object, var_value: Object, var_codingId: Object, var_code: Object, var_codedBy: Object, var_reason: Object, var_dictionaryName: Object, var_dictionaryVersion: Object, var_dateCoded: Object) =
        s"""
        | {
        | "studyKey":$var_studyKey,"siteName":$var_siteName,"siteId":$var_siteId,"subjectId":$var_subjectId,"subjectKey":$var_subjectKey,"formId":$var_formId,"formName":$var_formName,"formKey":$var_formKey,"revision":$var_revision,"recordId":$var_recordId,"variable":$var_variable,"value":$var_value,"codingId":$var_codingId,"code":$var_code,"codedBy":$var_codedBy,"reason":$var_reason,"dictionaryName":$var_dictionaryName,"dictionaryVersion":$var_dictionaryVersion,"dateCoded":$var_dateCoded
        | }
        """.stripMargin
}
